scipy/stats/_continuous_distns.py

```diff
@@ -2923,6 +2923,56 @@ def fit(self, data, *args, **kwds):
 gamma = gamma_gen(a=0.0, name='gamma')
 
 
+class gamma_gompertz_gen(rv_continuous):
+    """ A Gamma/Gompertz continuous random variable
+    
+    %(before_notes)s
+    
+    Notes
+    -----
+    The probability density function for `gamma_gompertz` is::
+    
+        gamma_gompertz.pdf(x, c, beta) = c * exp(x) * (beta**c) / ((beta - 1 + exp(x))**(c + 1))
+    
+    for ``x >= 0``, ``c > 0``, ``beta > 0``.
+    
+    `gamma_gompertz` takes ``c`` and ``beta`` as shape parameters
+    
+    %(after_notes)s
+    
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Gamma/Gompertz_distribution    
+    
+    %(example)s
+    """
+        
+    def _logsf(self, x, c, beta):
+        em1 = sc.expm1(x)
+        # np.where is used for dealing with the case that x is too large
+        # so that sc.expm1(x) becomes infinity.
+        # Since x is very large in that case, -sc.log1p(em1 / beta)
+        # approximately equals to -sc.log(np.exp(x) / beta) = np.log(beta) - x
+        return c * (np.where(np.isfinite(em1), -sc.log1p(em1 / beta),
```

Comment:
These conditionals should be tested, too. Ideally, provide a test with a set of parameters where it fails without np.where and passes (i.e. produces a correct result) with it. 

I also suspect there will be warnings. To avoid then, might want to use _lazywhere instead of np.where (see examples in other distributions in this file)